src/main/java/com/jmeter/websocket/plugin/endpoint/jetty/JettyWebsocketEndpoint.java

```diff
@@ -0,0 +1,158 @@
+package com.jmeter.websocket.plugin.endpoint.jetty;
+
+import com.google.common.base.Function;
+import com.google.common.base.Supplier;
+import com.jmeter.websocket.plugin.endpoint.WebsocketClient;
+import com.jmeter.websocket.plugin.endpoint.comsumers.WebsocketMessageProcessor;
+import com.jmeter.websocket.plugin.endpoint.jetty.converters.CookieManagerToCookieStoreConverter;
+import com.jmeter.websocket.plugin.endpoint.jetty.converters.HeadersToClientUpgradeRequestConverter;
+import com.jmeter.websocket.plugin.endpoint.jetty.converters.SampleResultToUpgradeListenerConverter;
+import org.apache.jmeter.protocol.http.control.CookieManager;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jorphan.logging.LoggingManager;
+import org.apache.log.Logger;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.eclipse.jetty.websocket.api.Session;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.jetty.websocket.client.io.UpgradeListener;
+
+import java.io.IOException;
+import java.net.CookieStore;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Future;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Suppliers.memoize;
+import static java.lang.Integer.toHexString;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class JettyWebsocketEndpoint implements WebsocketClient {
+
+    private static final Logger log = LoggingManager.getLoggerForClass();
+
+    private static Supplier<WebSocketClient> webSocketClientSupplier = webSocketClientSupplier();
+
+    private Supplier<JettyWebsocket> jettyWebsocketSupplier = jettyWebsocketSupplier();
+    private Function<CookieManager, CookieStore> cookieManagerToCookieStoreConverter = new CookieManagerToCookieStoreConverter();
+    private Function<Map<String, List<String>>, ClientUpgradeRequest> headersToClientUpgradeRequestConverter = new HeadersToClientUpgradeRequestConverter();
+    private Function<SampleResult, UpgradeListener> sampleResultToUpgradeListenerConverter = new SampleResultToUpgradeListenerConverter();
+
+    private Collection<WebsocketMessageProcessor> websocketMessageProcessors = new ArrayList<>();
+    private ConcurrentHashMap<URI, Session> sessions = new ConcurrentHashMap<>();
```

Comment:
Whole class looks a bit messy. What I'd do is to move session management to a separate object. Enpoint should depend on session manager responses. E. g. there are a few methods where you check that session is not null nor opened.